### ISSUE TITLE
October ASB 2018.10

### ISSUE THREAD

**Issue by MSe1969** [cm12-amami/discussion#42]
Like every 1st Monday of the month...

I am on vacation, so can only work on it starting mid of the month...
If you guys have time, please go ahead.
Pls. also cherry-pick the webview update from the 15.1 Lineage gerrit.

**Comment by Micha-Btz** [comment]
Summary:
**Framework**
- n/a CVE-2018-9490 | A-111274046[2] | EoP | Critical - libpac only adds test for v8, and v8 fixes functionality not yet introduced in this version
- n/a CVE-2018-9491 | A-111603051 | RCE | High - (builtin GCC 5 and above functionality)
- n/a CVE-2018-9492 | A-111934948 | EoP | High - 8.x and above
- [x] CVE-2018-9493 | A-111085900 [2][3] | ID | High - partly implemented; (DownloadProvider [2] n/a, as this would only apply, if the D/L search was enabled)
- n/a CVE-2018-9452 | A-78464361        [2] | DoS | Moderate - function/class not available

**Media framework**
- n/a CVE-2018-9473 | A-65484460 | RCE | Critical - 8.x and above
- n/a CVE-2018-9496 | A-110769924 | RCE | Critical - 9 only
- n/a CVE-2018-9497 | A-74078669 | RCE | Critical - no libmpeg2 in LP
- [ ] CVE-2018-9498 | A-78354855 | RCE | Critical - skia to be investigated
- n/a CVE-2018-9499 | A-79218474 | ID | High - function not available

**System**
- n/a CVE-2017-13283 | A-78526423 | RCE | Critical - No AVRC_PDU_LIST_PLAYER_APP_VALUES:
- n/a CVE-2018-9476 | A-109699112 | EoP | Critical - 8.x and above
- [x] CVE-2018-9504 | A-110216176 | RCE | Critical - implemented
- [x] CVE-2018-9501 | A-110034419 | EoP | High - implemented
- [x] CVE-2018-9502 | A-111936792 [2][3] | ID | High - implemented 
- [x] CVE-2018-9503 | A-80432928 [2][3] | ID | High - implemented, same as CVE-2018-9502 
- [x] CVE-2018-9505 | A-110791536 | ID | High - implemented
- [x] CVE-2018-9506 | A-111803925 | ID | High - implemented 
- [x] CVE-2018-9507 | A-111893951 | ID | High - implemented
- n/a CVE-2018-9508 | A-111936834 | ID | High - no smp_process_keypress_notification()
- [x] CVE-2018-9509 | A-111937027 | ID | High - implemented
- n/a CVE-2018-9510 | A-111937065 | ID | High - no smp_command_has_invalid_parameters()
- n/a CVE-2018-9511 | A-111650288 | DoS | High | - 9 only

**Kernel components**
- n/a CVE-2018-9513 | A-111081202* | EoP | High | - not found 
- n/a CVE-2018-9514 | A-111642636* | EoP | High | - we don't use sdcardfs
- n/a CVE-2018-9515 | A-111641492* | EoP | High | - we don't use sdcardfs

Pixel bulletins are not security relevant, only "Functional patches". 
Closing all patches that are not relevant for Android 5.

**Comment by Micha-Btz** [comment]
The fix for CVE-2018-9491 leading to compile problems, will try to find a fix.

```
The frameworks/av/media/ndk/NdkMediaCodec.cpp: In function 'AMediaCodecCryptoInfo* AMediaCodecCryptoInfo_new(int, uint8_t*, uint8_t*, cryptoinfo_mode_t, size_t*, size_t*)':
frameworks/av/media/ndk/NdkMediaCodec.cpp:417:78: error: '__builtin_mul_overflow' was not declared in this scope
     if (__builtin_mul_overflow(sizeof(size_t) * 2, numsubsamples, &cryptosize) ||
                                                                              ^
frameworks/av/media/ndk/NdkMediaCodec.cpp:418:90: error: '__builtin_add_overflow' was not declared in this scope
             __builtin_add_overflow(cryptosize, sizeof(AMediaCodecCryptoInfo), &cryptosize)) {

```

**Comment by Micha-Btz** [comment]
Sqlite patch are not willing to compile too. 
```
target Package: Galaxy4 (/home/mdomann/handy/build/cm-12.1/out/target/product/amami/obj/APPS/Galaxy4_intermediates/package.apk)
frameworks/base/core/java/android/database/sqlite/SQLiteQueryBuilder.java:397: error: cannot find symbol
            db.validateSql(unwrappedSql, cancellationSignal); // will throw if query is invalid
              ^
  symbol:   method validateSql(String,CancellationSignal)
  location: variable db of type SQLiteDatabase
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
```

**Comment by MSe1969** [comment]
Will Start having a look tomorrow, returning tonight from vacations

**Comment by MSe1969** [comment]
I've updated the summary:
1. marked n/a the n/a ones, so check mark would mean: "implemented" and unchecked "to be implemented or analyzed"
2. A couple of previously considered n/a I have "reset"; especially bluetooth. system/bt translates to external/bluetooth/bluedroid in LP and all .cc files in AOSP (C++) are .c files in LP, so all files do exist; it's the usual "fun" to backport C++ into regular C... // further: external/v8 in O equals external/chromium_org/v8 in LP, etc...
=> will have a look later to see, how far I get... - I know that in cm-14.1, they have difficulties with skia and wrt v8, they simply point to O-AOSP-v8 instead of backporting...

**Comment by MSe1969** [comment]
FYI: I'll wipe your commit https://github.com/cm12-amami/android_external_bluetooth_bluedroid/commit/e0b4ea7cc7e1d13aa3ee1a541e16feec628a5a9d via "force-push", because the patch is already present, see here: https://github.com/cm12-amami/android_external_bluetooth_bluedroid/commit/b018333a49d33e1fa24cb9b99df84a067d16177c
It seems that the commit was somehow reworked or comes from a different tag, as the "android_errorWriteLog" part is not present any more in the AOSP link, but this is, in my opinion, of minor importance.

**Comment by MSe1969** [comment]
CVE-2018-9491: __builtin_mul_overflow() is abuilt-in overflow check function, available as of GCC 5 (https://gcc.gnu.org/gcc-5/changes.html) or clang, both *not* used in Android 5 build tree, so I would assume N/A. Thoughts?

**Comment by MSe1969** [comment]
**Status:**
Implemented everything (compiles fine), except for
CVE-2018-9490  => N/A
CVE-2018-9491  => Moved to "open CVE's issue"; commit wiped
CVE-2018-9498  => Moved to "open CVE's issue", but still working on it
CVE-2018-9501  => implemented

**Comment by MSe1969** [comment]
@hahnjo : do you maybe have an idea with regards to CVE-2018-9491 and the use of `__builtin_mul_overflow()` ?

**Comment by MSe1969** [comment]
Test build is running now, let's ship this way (if all works out fine), and we'll work on skia in parallel

**Comment by MSe1969** [comment]
@Micha-Btz @hahnjo  
**Okay, my test build runs fine - if I read no objections from your side, I'll ship tomorrow night.**

CVE-2018-9491 and CVE-2018-9498 are deferred.
For CVE-2018-9498, I have created a branch cm-12.1-tmp in external/skia with my attempt to backport. Unfortunately, it won't compile yet, see here:
```
target thumb C++: libskia <= external/skia/src/ports/SkTLS_pthread.cpp
In file included from external/skia/include/core/SkFixed.h:11:0,
                 from external/skia/include/core/SkScalar.h:11,
                 from external/skia/include/core/SkString.h:13,
                 from external/skia/include/core/SkOSFile.h:15,
                 from external/skia/src/ports/SkOSFile_posix.cpp:8:
external/skia/src/utils/SkTFitsIn.h: In instantiation of 'bool SkTFitsIn(S) [with D = unsigned int; S = long long int]':
external/skia/src/ports/SkOSFile_posix.cpp:55:42:   required from here
external/skia/src/utils/SkTFitsIn.h:203:59: error: size of array is negative
     SK_COMPILE_ASSERT(std::numeric_limits<S>::is_integer, SkTFitsIn_source_must_be_integer);
                                                           ^
external/skia/include/core/SkTypes.h:212:13: note: in definition of macro 'SK_COMPILE_ASSERT'
             msg[static_cast<bool>(expr) ? 1 : -1] SK_UNUSED
             ^
build/core/binary.mk:619: die Regel für Ziel „/home/markus/out-android/lin-12.1/target/product/amami/obj/SHARED_LIBRARIES/libskia_intermediates/src/ports/SkOSFile_posix.o“ scheiterte
```
Maybe you have any idea?

**Comment by MSe1969** [comment]
shipped. Any comments before I merge the test branch into stable?

**Comment by Micha-Btz** [comment]
Sorry I'm ill, I didn't have test something.

**Comment by MSe1969** [comment]
Test branch merged into stable branch, so closing this issue now.
CVE-2018-9491 and CVE-2018-9498 are deferred and added to the "open CVEs" repo

**Comment by foresto** [comment]
> CVE-2018-9493 | A-111085900 [2][3] | ID | High - partly implemented; (DownloadProvider [2] n/a, as this would only apply, if the D/L search was enabled)

Do we have authoritative confirmation that it doesn't apply? Even if interactive search is disabled, couldn't a malicious app find a way to trigger a search without user interaction? [The CVE](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2018-9493) says, "User interaction is not needed for exploitation." This suggests to me that we should be careful about assuming that cm-12.1 is unaffected.
